pre-commit.softHSM.Dockerfile

```diff
@@ -0,0 +1,21 @@
+#
+# Docker Image for pre-commit.sh to run all tests.
+# softHSM tests require softHSM and OpenSC.
+# Gradle required for code build.
+#
+FROM gradle:jdk11
```

Comment:
This Dockerfile is the same as the runtime one, apart from this line. Is there any way to reduce the replication?

Reply:
I think you could reduce the two files to one. You could set an `ARG` before the `FROM` that you set to the image you want to base them off (either `openjdk:11-jre-slim`, or  `gradle:jdk11`), and then use that arg in the `FROM` directive. You just need to set the arg to the right thing in the docker-compose and gradle-compose files.